### ISSUE TITLE
openapi-fetch `Type instantiation is excessively deep and possibly infinite.`

### ISSUE THREAD

**Issue by vovsemenv** [openapi-ts/openapi-typescript#1882]
I have the code with 2 api calls - one with  "openapi-fetch" other with "openapi-react-query"
but only "openapi-react-query" have actual error on paths that didn't exist, 
"openapi-fetch"generates the `Type instantiation is excessively deep and possibly infinite.`

Because of that vscode autocomplete didn't work in case with "openapi-fetch"

```ts
import type { paths } from "app/typings/__generated/api";

import createFetchClient from "openapi-fetch";
import createClient from "openapi-react-query";

export const fetchClient = createFetchClient<paths>({});
export const $api = createClient(fetchClient);

fetchClient.GET("/api/d");
// ^ Type instantiation is excessively deep and possibly infinite.
$api.useQuery("get", "/api/sdf");
// ^ Argument of type '"/api/sdf"' is not assignable to parameter of type 'PathsWithMethod<paths, "get">'

fetchClient.GET("/api/accessgroup");
// ^ Existing url works fine
$api.useQuery("get", "/api/accessgroup");
// ^ Existing url works fine

``

**Comment by drwpow** [comment]
As you discerned, this error does appear when there’s an actual error. It’s tough to control the output for what TypeScript shows beyond throwing an error. There are no known instances where this happens for a legitimate path. But agreed the error could be more helpful.

Flagged as an “enhancement” as in “this error message should be improved” although we are limited by what TypeScript allows (since it’s the one that shows that error). PRs are welcome for anyone that can improve this! (And bonus points if it can be tested)

**Comment by vovsemenv** [comment]
Yes i understand that😅 Just really strange to see that the same thing works in react-query version

**Comment by rasmushallstrom** [comment]
I have same problem in vscode no autocomplete on paths and getting "Type instantiation is excessively deep and possibly infinite."

But it works fine in webstorm.

**Comment by rasmushallstrom** [comment]
I am testing a large YAML file (GitHub API) with over 900 paths. When using openapi-fetch in VS Code, I encounter the same issue:

client.GET('');
// Error: Type instantiation is excessively deep and possibly infinite.

Observations:
This issue occurs consistently in VS Code.
WebStorm handles the same file without any problems.
Steps to Reproduce:

Use a large OpenAPI YAML file (e.g., GitHub API with 900+ paths).
Generate TypeScript types using openapi-typescript.
Attempt to use client.GET('') in VS Code.
Expected Behavior: VS Code should handle the type instantiation without errors, similar to WebStorm.

**Comment by ashrhmn** [comment]
> I am testing a large YAML file (GitHub API) with over 900 paths. When using openapi-fetch in VS Code, I encounter the same issue:
> 
> client.GET(''); // Error: Type instantiation is excessively deep and possibly infinite.
> 
> Observations: This issue occurs consistently in VS Code. WebStorm handles the same file without any problems. edit: tried createPathBasedClient instead and it works with no problem in vscode
> 
> Steps to Reproduce: Use a large OpenAPI YAML file (e.g., GitHub API with 900+ paths). Generate TypeScript types using openapi-typescript. Attempt to use client.GET('') in VS Code. Expected Behavior: VS Code should handle the type instantiation without errors, similar to WebStorm.

+1 🙏

**Comment by vovsemenv** [comment]
> I also ran into this issue with `openapi-fetch` and the standard client. I tried using `wrapAsPathBasedClient` and it seems that the url type checking is working with the proxy:
> 
> > Property '/posts/{post_id}' does not exist on type 'PathBasedClient<paths, `${string}/${string}`>'. Did you mean '/blogposts/{post_id}'?

THANK YOU!!! for some reason vscode works much faster with the path Based client!!!!!

**Comment by Diggsey** [comment]
I have this issue too... The non-path-based client just does not work...
